### PR TITLE
Task ys/j44 create episode page

### PR DESCRIPTION
**Pull request by yaroslav207**
image

### CODE REVIEW

File: frontend/src/common/enums/app/app-route.enum.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ enum AppRoute {
   PODCAST_EDIT = '/podcasts/edit',
   PODCAST_EDIT_$ID = '/podcasts/edit/:id?',
   PODCAST_$ID_EPISODE_$ID  = '/podcasts/:podcastId/episode/:episodeId',
+  EPISODE_CREATE = '/podcasts/:podcastId/episode/create',
```
Comment (ghost) [inline frontend/src/common/enums/app/app-route.enum.ts]:
```suggestion
  PODCAST_$ID_EPISODE_EDIT_$ID = '/podcasts/:podcastId/episode/edit/:episodeId?',
```

File: frontend/src/components/common/select/select.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  FieldErrors,
+  useController,
+  Path,
+  FieldValues,
+} from 'react-hook-form';
+
+import styles from './styles.module.scss';
+
+type Props = {
+  selectItems: Array<string>,
```
Comment (ghost) [inline frontend/src/components/common/select/select.tsx]:
```suggestion
  options: string[],
```

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          type={InputType.TEXT}
+          label="Description"
+          placeholder="Enter episode description"
+          name={EpisodeCreatePayloadKey.DESCRIPTION}
+          control={control}
+          errors={errors}
+        />
+        <Select
+          selectItems={[EpisodeType.PUBLIC, EpisodeType.PRIVATE, EpisodeType.UNLISTED]}
+          label={'Type'}
```
Comment (ghost) [inline frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]:
```suggestion
          label="Type"
```

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        <Input
+          type={InputType.TEXT}
+          label="Description"
+          placeholder="Enter episode description"
+          name={EpisodeCreatePayloadKey.DESCRIPTION}
+          control={control}
+          errors={errors}
+        />
+        <Select
+          selectItems={[EpisodeType.PUBLIC, EpisodeType.PRIVATE, EpisodeType.UNLISTED]}
```
Comment (ghost) [inline frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]:
```suggestion
         const episodeTypes = Object.values(EpisodeType);

          selectItems={episodeTypes}
```

File: frontend/src/components/configurate-episode/configurate-episode.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { EpisodeCreatePayload } from 'common/types/types';
+
+import { PageParams } from './common/types/types';
+import { CreateEpisodeForm } from './components/components';
+
+const ConfigurateEpisode: React.FC = () => {
+  const { episodeId } = useParams<PageParams>();
+  const dispatch = useDispatch();
+
+  const isEdit = Boolean(episodeId);
```
Comment (ghost) [inline frontend/src/components/configurate-episode/configurate-episode.tsx]:
```suggestion
```
for not don't handle it

File: frontend/src/store/episode/actions.ts [hunk 6]
```diff
@@ -10,4 +10,12 @@ const loadEpisode = createAsyncThunk<Episode, number, AsyncThunkConfig>
   return episode;
 });
 
-export { loadEpisode };
+const createEpisode = createAsyncThunk<Episode, EpisodeCreatePayload, AsyncThunkConfig>
```
Comment (ghost) [inline frontend/src/store/episode/actions.ts]: you need to create your own reducer, actions and etc called `configurateEpisode`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/store/configurate-episode/common.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+enum ActionType {
+  CREATE_EPISODE = 'episodes/create-episode',
```
Comment (ghost) [inline frontend/src/store/configurate-episode/common.ts]:
```suggestion
  CREATE_EPISODE = 'configurate-episode/create-episode',
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]
👍

**Pull request merged**